lib/regular_expression/bytecode.rb

```diff
@@ -38,7 +38,7 @@ def self.compile(nfa)
           builder.mark_label(label[state, index])
 
           if state.transitions.length > 1 && index != state.transitions.length - 1
-            builder.push(Insns::PushIndex.new)
+            builder.push(Insns::PushIndex.new(n_of_transitions))
```

Comment:
This makes sense for other backends as well - the x86 backend could use simple `mov` instructions rather than stack operations, using this. It'll reduce false data dependencies.